src/common/meta/src/wal/kafka/topic_manager.rs

```diff
@@ -0,0 +1,147 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::HashSet;
+
+use rand::seq::SliceRandom;
+use rskafka::client::ClientBuilder;
+use snafu::{ensure, OptionExt, ResultExt};
+
+use crate::error::{
+    BuildKafkaClientSnafu, BuildKafkaCtrlClientSnafu, CreateKafkaTopicSnafu, DeserKafkaTopicsSnafu,
+    InvalidNumTopicsSnafu, MissingKafkaOptsSnafu, Result, SerKafkaTopicsSnafu,
+    TooManyCreatedKafkaTopicsSnafu,
+};
+use crate::kv_backend::KvBackendRef;
+use crate::rpc::store::PutRequest;
+use crate::wal::kafka::topic_selector::{build_topic_selector, TopicSelectorRef};
+use crate::wal::kafka::{KafkaOptions, TopicSelectorType};
+
+pub type Topic = String;
+
+const METASRV_CREATED_TOPICS_KEY: &str = "metasrv_created_topics";
+const CREATE_TOPIC_TIMEOUT: i32 = 5_000; // 5,000 ms.
+
+pub struct TopicManager {
+    topic_pool: Vec<Topic>,
+    topic_selector: TopicSelectorRef,
+}
+
+impl TopicManager {
+    pub async fn try_new(
+        kafka_opts: Option<&KafkaOptions>,
+        kv_backend: &KvBackendRef,
+    ) -> Result<Self> {
+        let opts = kafka_opts.context(MissingKafkaOptsSnafu)?;
+        let topic_pool = build_topic_pool(opts, kv_backend).await?;
```

Comment:
Do we really need to allocate these resources in `try_new` method?

Reply:
👍  We can provide a `start` function to do this.

Reply:
The initialization of `TopicManager` is split into `new` and `try_init` methods.